### PR TITLE
Support non-spatial mode in BatchNormalization

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: Older opset spec of BatchNormalization supports non-spatial mode of the operator supported by some frameworks. This change supports that mode of operation.

**Motivation and Context**
Keep ORT backwards compatible

### CODE REVIEW

File: onnxruntime/test/onnx/main.cc [hunk 1]
```diff
@@ -532,7 +532,6 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
 #endif
 
 #ifdef USE_CUDA
-  broken_tests.insert({"mxnet_arcface", "result mismatch"});
```
Comment (hariharans29) [inline onnxruntime/test/onnx/main.cc]: Added to the exclusion list once above

File: csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs [hunk 2]
```diff
@@ -319,7 +319,7 @@ private void TestMultiThreads()
         private static Dictionary<string, string> GetSkippedModels()
         {
             var skipModels = new Dictionary<string, string>() {
-                { "mxnet_arcface", "Model not supported by CPU execution provider" } ,
+                { "mxnet_arcface", "Model is an invalid ONNX model",
```
Comment (hariharans29) [inline csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs]: See reply here on why it is invalid: 
https://github.com/onnx/models/issues/156

File: onnxruntime/core/providers/cpu/nn/batch_norm_helper.h [hunk 3]
```diff
@@ -14,48 +14,92 @@ class BatchNormHelper {
...
-    constexpr int kNumInputMeanDimensions = 1;
-    constexpr int kNumInputVarianceDimensions = 1;
+    int kNumInputScaleDimensions = (is_spatial ? 1 : num_feature_dims + 1);
+    int kNumInputBiasDimensions = (is_spatial ? 1 : num_feature_dims + 1);
+    int kNumInputMeanDimensions = (is_spatial ? 1 : num_feature_dims + 1);
+    int kNumInputVarianceDimensions = (is_spatial ? 1 : num_feature_dims + 1);
     //constexpr int kMinCudaNumDims = 4;
     //constexpr int kMaxCudaNumDims = 5;
 
-    if (X->Shape().GetDims().empty()) {
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/nn/batch_norm_helper.h]: Removed this check in favor of the more stricter check in line 20.

File: onnxruntime/core/providers/cpu/nn/batch_norm_helper.h [hunk 4]
```diff
@@ -14,48 +14,92 @@ class BatchNormHelper {
                                        const Tensor* scale,
                                        const Tensor* B,
                                        const Tensor* mean,
-                                       const Tensor* var) {
+                                       const Tensor* var,
+                                       bool is_spatial = true) {
+    const auto& x_dims = X->Shape().GetDims();
+    if (x_dims.size() < 2) {
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/nn/batch_norm_helper.h]: Even though, the spec doesn't explicitly state it, it makes sense if the minimum rank of 'X' is atleast 3 - N,C, and atleast one data dimension (D1). However, when we impose that, it seems to break some successfully running models that have input rank 2 - [N, C]. The implicit shape is [N, C, 1] in those cases and the results seem ok. Hence, modifying the minimum rank criteria to just 2. This is okay to add as in the next step we parse the number of channels in the index position 1.

File: onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc [hunk 5]
```diff
@@ -23,19 +23,27 @@ void TestBatchNorm(const InputDataMap& input_data_map,
...
+  }
   test.AddInput<float>("X", input_shapes_map.at("X"), input_data_map.at("X"));
   test.AddInput<float>("scale", input_shapes_map.at("scale"), input_data_map.at("scale"));
   test.AddInput<float>("B", input_shapes_map.at("B"), input_data_map.at("B"));
   test.AddInput<float>("mean", input_shapes_map.at("mean"), input_data_map.at("mean"));
   test.AddInput<float>("var", input_shapes_map.at("var"), input_data_map.at("var"));
   test.AddOutput<float>("output", expected_output_shape, expected_output);
-  test.Run(expect_result, err_str, {kTensorrtExecutionProvider});// Weight as input is not supported by TensorRT
+  // Weight as input is not supported by TensorRT and spatial == 0 is not supported by Nuphar
+  std::unordered_set<std::string> excluded_eps = {kCudaExecutionProvider, kTensorrtExecutionProvider};
```
Comment (HectorSVC) [inline onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc]: >kCudaExecutionProvider [](start = 50, length = 22)

spatial == 0 should be supported by CUDA

Comment (hariharans29) [reply onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc]: Thanks for catching. Added that incorrectly.

Comment (hariharans29) [reply onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc]: Fixed
